### PR TITLE
fix(neovim): add in update-aternatives support

### PR DESCRIPTION
**Pull request by erenfro**
This adds in update-alternatives so neovim, view, and vimdiff can be registered and used in defaults.

### CODE REVIEW

**Comment by Elsie19** [comment]
Can this be done for `neovim-git` as well?

File: packages/neovim/neovim.pacscript [hunk 1]
```diff
@@ -1,11 +1,20 @@
 pkgname="neovim"
 pkgver="0.10.0"
-source=("https://github.com/neovim/neovim/archive/refs/tags/v${pkgver}.tar.gz")
+pkgrel="1"
```
Comment (Elsie19) [inline packages/neovim/neovim.pacscript:3]:
```suggestion
pkgrel="2"
```

**Review by Elsie19 — CHANGES_REQUESTED** [review]

File: packages/neovim/neovim.pacscript [hunk 2]
```diff
@@ -1,23 +1,82 @@
...
+  install -Dm755 "${srcdir}/ex" -t "${pkgdir}/usr/libexec/${pkgname}"
+  install -Dm755 "${srcdir}/rview" -t "${pkgdir}/usr/libexec/${pkgname}"
+  install -Dm755 "${srcdir}/rvim" -t "${pkgdir}/usr/libexec/${pkgname}"
+  install -Dm755 "${srcdir}/view" -t "${pkgdir}/usr/libexec/${pkgname}"
+  install -Dm755 "${srcdir}/vimdiff" -t "${pkgdir}/usr/libexec/${pkgname}"
+}
+
+post_install() {
+  set -e
+
```
Comment (Elsie19) [inline packages/neovim/neovim.pacscript:47]:
```suggestion
```

File: packages/neovim/neovim.pacscript [hunk 3]
```diff
@@ -1,23 +1,82 @@
...
+  update-alternatives --install /usr/bin/rview rview "/usr/libexec/${pkgname}/rview" 30
+  update-alternatives --install /usr/bin/vi vi /usr/bin/nvim 30
+  update-alternatives --install /usr/bin/vim vim /usr/bin/nvim 30
+  update-alternatives --install /usr/bin/view view "/usr/libexec/${pkgname}/view" 30
+  update-alternatives --install /usr/bin/vimdiff vimdiff "/usr/libexec/${pkgname}/vimdiff" 30
+}
+
+pre_remove() {
+  set -e
+
```
Comment (Elsie19) [inline packages/neovim/neovim.pacscript:62]:
```suggestion
```

**Review by Elsie19 — CHANGES_REQUESTED** [review]

File: packages/neovim/neovim.pacscript (hunk 2 above)
Comment (erenfro) [reply packages/neovim/neovim.pacscript:47]: And why would you do this? This is actually intentionally set in this specific case.

File: packages/neovim/neovim.pacscript (hunk 3 above)
Comment (erenfro) [reply packages/neovim/neovim.pacscript:62]: This too. This is intentional, and also based on the original neovim preinst control script.

**Pull request merged**